omnigibson/objects/stateful_object.py

```diff
@@ -521,8 +520,10 @@ def _dump_state(self):
         # Also add non-kinematic states
         non_kin_states = dict()
         for state_type, state_instance in self._states.items():
-            if state_instance.stateful:
+            try:
                 non_kin_states[get_state_name(state_type)] = state_instance.dump_state(serialized=False)
+            except NotImplementedError:
+                pass
```

Comment:
Do we absolutely need this try/except? why is the `stateful` check not good enough?

Reply:
I agree this is not ideal. But since we have got rid of `stateful` in this PR, this seems to be the only way to do it.

Reply:
Revert this with the state size